frame/bounties/src/migrations_4_0_0.rs

```diff
@@ -0,0 +1,174 @@
+	Funded,
+	/// A curator has been proposed by the `ApproveOrigin`. Waiting for acceptance from the curator.
+	CuratorProposed {
+		/// The assigned curator of this bounty.
+		curator: AccountId,
+	},
+	/// The bounty is active and waiting to be awarded.
+	Active {
+		/// The curator of this bounty.
+		curator: AccountId,
+		/// An update from the curator is due by this block, else they are considered inactive.
+		update_due: BlockNumber,
+	},
+	/// The bounty is awarded and waiting to released after a delay.
+	PendingPayout {
+		/// The curator of this bounty.
+		curator: AccountId,
+		/// The beneficiary of the bounty.
+		beneficiary: AccountId,
+		/// When the bounty can be claimed.
+		unlock_at: BlockNumber,
+	},
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
+pub struct OldBounty<AccountId, Balance, BlockNumber> {
+	/// The account proposing it.
+	proposer: AccountId,
+	/// The (total) amount that should be paid if the bounty is rewarded.
+	value: Balance,
+	/// The curator fee. Included in value.
+	fee: Balance,
+	/// The deposit of curator.
+	curator_deposit: Balance,
+	/// The amount held on deposit (reserved) for making this proposal.
+	bond: Balance,
+	/// The status of this bounty.
+	status: BountyStatus<AccountId, BlockNumber>,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
+pub struct Bounty<AccountId, Balance, BlockNumber> {
+	/// The account proposing it.
+	proposer: AccountId,
+	/// The (total) amount that should be paid if the bounty is rewarded.
+	value: Balance,
+	/// The curator fee. Included in value.
+	fee: Balance,
+	/// The deposit of curator.
+	curator_deposit: Balance,
+	/// The amount held on deposit (reserved) for making this proposal.
+	bond: Balance,
+	/// The status of this bounty.
+	status: BountyStatus<AccountId, BlockNumber>,
+	/// active Subbounty count
+	active_subbounty_count: BountyIndex,
+}
+
+struct __Bounties;
+impl frame_support::traits::StorageInstance for __Bounties {
+	fn pallet_prefix() -> &'static str { "Treasury" }
+	const STORAGE_PREFIX: &'static str = "Bounties";
+}
+#[allow(type_alias_bounds)]
+type Bounties<T: V3ToV4> = StorageMap<
+								__Bounties,
+								Twox64Concat,
+								BountyIndex,
+								Bounty<T::AccountId, T::Balance, T::BlockNumber>
+							>;
+
+/// Apply all of the migrations from 3_0_0 to 4_0_0.
+///
+/// ### Warning
+///
+/// This code will **ONLY** check that the storage version is less than or equal to 2_0_0.
+/// Further check might be needed at the user runtime.
+///
+/// Be aware that this migration is intended to be used only for the mentioned versions. Use
+/// with care and run at your own risk.
+pub fn apply<T: V3ToV4>( ) -> Weight {
+
+	let maybe_storage_version = crate_to_pallet_version!();
+
+	log::info!(
+		target: "runtime::pallet-bounties",
+		"Running migration for pallet-bounties with storage version {:?}",
+		maybe_storage_version,
+	);
+
+	if maybe_storage_version <= PalletVersion::new(4, 0, 0) {
```

Comment:
shouldn't it be `<` ? if version is 4.0.0 we don't want to run the migration.

also we should bump the version in the Cargo.toml

Reply:
> also we should bump the version in the Cargo.toml

I don't know what we should do here: bump major version every time we need migration so that we can make use of version when doing migration in polkadot (because polkadot is using substrate master).

Or should we just avoid to check on the version here and do the `apply` at the runtime level without any check on versions

Reply:
Implemented the latter suggestion of avoiding version check in `apply()` method.